src/lib/services/searchUtilActions/SubmodelSearcher.ts

```diff
@@ -0,0 +1,109 @@
+import { Reference, Submodel } from '@aas-core-works/aas-core3.0-typescript/dist/types/types';
+import { SubmodelDescriptor } from 'lib/types/registryServiceTypes';
+import { ISubmodelRepositoryApi } from 'lib/api/basyx-v3/apiInterface';
+import { SubmodelRepositoryApi } from 'lib/api/basyx-v3/api';
+import { mnestixFetch } from 'lib/api/infrastructure';
+import { ISubmodelRegistryServiceApiInterface } from 'lib/api/submodel-registry-service/ISubmodelRegistryServiceApiInterface';
+import { SubmodelRegistryServiceApi } from 'lib/api/submodel-registry-service/submodelRegistryServiceApi';
+import { ApiResponseWrapper, ApiResultStatus, wrapErrorCode } from 'lib/util/apiResponseWrapper/apiResponseWrapper';
+import { RepositorySearchService } from 'lib/services/repository-access/RepositorySearchService';
+
+export class SubmodelSearcher {
+    private constructor(
+        protected readonly submodelRepositoryClient: ISubmodelRepositoryApi,
+        protected readonly submodelRegistryClient: ISubmodelRegistryServiceApiInterface,
+        protected readonly multipleDataSource: RepositorySearchService,
+    ) {}
+
+    static create(): SubmodelSearcher {
+        const multipleDataSource = RepositorySearchService.create();
+        const submodelRepositoryClient = SubmodelRepositoryApi.create(
+            mnestixFetch(),
+            undefined,
+            process.env.SUBMODEL_REPO_API_URL ?? process.env.AAS_REPO_API_URL,
+        );
+        const submodelRegistryClient = SubmodelRegistryServiceApi.create(
+            process.env.SUBMODEL_REPO_API_URL ?? process.env.AAS_REPO_API_URL,
```

Comment:
This might should be the `SUBMODEL_REGISTRY_API_URL` 🤔

Reply:
good catch!